_rules/iframe-not-focusable-has-no-interactive-content-akn7bn.md

```diff
@@ -28,13 +28,17 @@ acknowledgments:
 
 ## Applicability
 
-This rule applies to any `iframe` element that [contains](#akn7bn:contain) at least one element for which all the following are true:
+This rule applies to any `iframe` element that is not [inert](#akn7bn:inert) and that [contains](#akn7bn:contain) at least one element for which all the following are true:
 
 - the element is [visible][]; and
 - the element is part of the [sequential focus navigation order][] of the `iframe`'s [document][].
 
 An element is <dfn id="akn7bn:contain">contained</dfn> in a [nested browsing context][] if its [owner document][] is the [container document][] of the [nested browsing context][].
 
+An `iframe` element is <dfn id="akn7bn:inert">inert</dfn> if:
+- it has an `inert` [attribute value][] of true; or
+- an element which is not the iframe itself and that is not [contained](#akn7bn:contain) in it behaves as a [modal][], making the iframe inoperable.
```

Comment:
```suggestion
- an element which is not the iframe itself and that is not [contained](#akn7bn:contain) in it makes the iframe and all of its contents [inoperable][].  (For example: a modal.)
```
Does that improve the situation at all?

Reply:
What I meant by modal dialogs was the actual `<dialog>` element with `showModal()` called on it. That programmatically makes all other content on the page inert. I think you could define it like this:

> The element is in a page with a `dialog` element with the [`is-modal`](https://html.spec.whatwg.org/multipage/interactive-elements.html#is-modal) flag set to `true`, and is not a descendant of that `dialog` element.


I'm not sure we should try for a generic modal definition. I don't know how you'd do that in a way that's objective.

Reply:
Then, we are indeed going for [blocked by a modal](https://html.spec.whatwg.org/multipage/interaction.html#blocked-by-a-modal-dialog):
> A [Document](https://html.spec.whatwg.org/multipage/dom.html#document) document is blocked by a modal dialog subject if subject is the topmost [dialog](https://html.spec.whatwg.org/multipage/interactive-elements.html#the-dialog-element) element in document's [top layer](https://fullscreen.spec.whatwg.org/#top-layer). While document is so blocked, every node that is [connected](https://dom.spec.whatwg.org/#connected) to document, with the exception of the subject element and its [flat tree](https://drafts.csswg.org/css-scoping/#flat-tree) descendants, must become [inert](https://html.spec.whatwg.org/multipage/interaction.html#inert).

> subject can additionally become [inert](https://html.spec.whatwg.org/multipage/interaction.html#inert) via the [inert](https://html.spec.whatwg.org/multipage/interaction.html#the-inert-attribute) attribute, but only if specified on subject itself (i.e., subject escapes inertness of ancestors); subject's [flat tree](https://drafts.csswg.org/css-scoping/#flat-tree) descendants can become [inert](https://html.spec.whatwg.org/multipage/interaction.html#inert) in a similar fashion.

> Note: The [dialog](https://html.spec.whatwg.org/multipage/interactive-elements.html#the-dialog-element) element's [showModal()](https://html.spec.whatwg.org/multipage/interactive-elements.html#dom-dialog-showmodal) method causes this mechanism to trigger, by [adding](https://fullscreen.spec.whatwg.org/#top-layer-add) the [dialog](https://html.spec.whatwg.org/multipage/interactive-elements.html#the-dialog-element) element to its [node document](https://dom.spec.whatwg.org/#concept-node-document)'s [top layer](https://fullscreen.spec.whatwg.org/#top-layer).